server/src/main/java/org/opensearch/search/ResourceType.java

```diff
@@ -10,21 +10,25 @@
 
 import org.opensearch.common.annotation.PublicApi;
 import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.tasks.Task;
 
 import java.io.IOException;
+import java.util.function.Function;
 
 /**
  * Enum to hold the resource type
  */
 @PublicApi(since = "2.x")
 public enum ResourceType {
-    CPU("cpu"),
-    MEMORY("memory");
+    CPU("cpu", task -> task.getTotalResourceStats().getCpuTimeInNanos()),
+    MEMORY("memory", task -> task.getTotalResourceStats().getMemoryInBytes());
```

Comment:
This can probably be taken as a future optimization, but just at a glance this logic looks a little clunky. `task.getTotalResourceStats()` returns a new `TaskResourceUsage` object which should contain both the CPU and the memory resource utilization. That means we are reconstructing that object on each  `getResourceUsage` call and since you have a triple for loop calling this in `QueryGroupResourceUsageTrackerService` I'm not sure if that is negligible. It's probably better to just implement a `task.getTotalCPUResourceStats()` method or something like that.

Reply:
ya very good call out, addressed, thanks !